test/lib/duck/resource_checks.go

```diff
@@ -43,17 +39,21 @@ const (
 // it is done, returns an error or timeout.
 func WaitForResourceReady(dynamicClient dynamic.Interface, obj *resources.MetaResource) error {
 	return wait.PollImmediate(interval, timeout, func() (bool, error) {
-		untyped, err := GetGenericObject(dynamicClient, obj, &duckv1beta1.KResource{})
-		return isResourceReady(untyped, err)
+		return checkResourceReady(dynamicClient, obj)
 	})
 }
 
+func checkResourceReady(dynamicClient dynamic.Interface, obj *resources.MetaResource) (bool, error) {
+	untyped, err := GetGenericObject(dynamicClient, obj, &kresourceForReadiness{})
```

Comment:
I wonder if it's cleaner just to have this check the obj coming in and if it's a Pod, fetch it as a pod instead of treating it as a duck?

Reply:
:+1: That is what I would do.